src/standard/expr.ml

```diff
@@ -1855,7 +1867,7 @@ module Term = struct
         with_cache ~cache:(Hashtbl.create 13) (fun (i, j, n) ->
             mk' ~builtin:(Builtin.Bitv_extract (i, j))
               (Format.asprintf "bitv_extract_%d_%d" i j) []
-              [Ty.bitv n] (Ty.bitv (i - j + 1))
+              [Ty.bitv n] (Ty.bitv (j - i + 1))
```

Comment:
This looks like quite a breaking change: is the current really incorrect ? (which seems surprising given that there were no errors related to bitv extraction when benchmarking the smtlib), or did you just misunderstand the order of the arguments ? (which is admittedly very easy)

In any case, if this change is needed, it should be in a separate PR, with adequate test cases (particularly smtlib test cases).

Reply:
Yes, this was wrong. It's because in Alt-Ergo the extract operator takes the indexes of the extraction from start to end, while in Smt-Lib, it's the opposite. I fixed this and just inverted them in [src/typecheck/bitv.ml#L37](https://github.com/hra687261/dolmen/blob/ae-builtins/src/typecheck/bitv.ml#L37).